src/pages/organisationUnits/form/ImageField.tsx

```diff
@@ -0,0 +1,110 @@
+import { useDataEngine } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import {
+    FileInput,
+    FileInputChangeHandler,
+    FileList,
+    FileListItem,
+    Label,
+} from '@dhis2/ui'
+import React from 'react'
+import { Field as FieldRFF, useField } from 'react-final-form'
+
+export function ImageField() {
+    const dataEngine = useDataEngine()
+    const fieldName = 'image'
+    const { input } = useField(fieldName)
+    const currentFile: { name?: string; id?: string } | undefined =
+        input.value || undefined
+
+    const uploadFile = async (fileToUpload: File) => {
+        const ADD_NEW_FILE_RESOURCE_MUTATION = {
+            resource: 'fileResources',
+            type: 'create',
+            data: fileToUpload,
+        } as const
+        const fileToUploadDetails = {
+            name: fileToUpload.name,
+            size: fileToUpload.size,
+        }
+        updateInputValue({
+            ...fileToUploadDetails,
+            id: undefined,
+            error: undefined,
+        })
+        try {
+            const postResponse = (await dataEngine.mutate(
+                ADD_NEW_FILE_RESOURCE_MUTATION,
+                {
+                    variables: { file: fileToUpload },
+                }
+            )) as {
+                response: {
+                    fileResource: { id: string; storageStatus: string }
+                }
+            }
+            updateInputValue({
+                ...fileToUploadDetails,
+                id: postResponse.response.fileResource.id,
+                error: undefined,
+            })
+        } catch (e) {
+            updateInputValue({
+                ...fileToUploadDetails,
+                id: undefined,
+                error: (e as Error | string).toString(),
+            })
+        }
+    }
+    const updateInputValue = (newInputValue: {
+        name: string
+        size: number
+        id?: string
+        error?: string
+    }) => {
+        input.onChange(newInputValue)
+        input.onBlur()
+    }
+    const deleteFile = () => {
+        input.onChange(undefined)
+        input.onBlur()
+    }
+    const handleChange: FileInputChangeHandler = ({ files }) => {
+        const newFile = files[0]
+        if (newFile instanceof File) {
+            uploadFile(newFile)
+        }
+    }
+
+    return (
+        <FieldRFF<string | undefined> name={fieldName}>
```

Comment:
We don't need this. This will create a new `listener` for the field - and we already have `useField` above.